test_specific_improvements.py

```
@@ -0,0 +1,394 @@
+    """Test enhanced file format validation and corruption detection."""
+    print("🔍 Testing File Format Validation...")
+    
+    # Test 1: Corrupted PDF detection
+    with tempfile.NamedTemporaryFile(suffix='.pdf', delete=False) as tmp:
+        tmp.write(b'This is not a PDF file')
+        corrupted_path = Path(tmp.name)
+    
+    try:
+        try:
+            pipeline._validate_file_format(corrupted_path)
+            assert False, "Should have detected corrupted PDF"
```

Ruff (0.11.9)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)